projects/wave-core-new/src/lib/datasets/upload/upload.component.ts

```diff
@@ -0,0 +1,188 @@
+                    "name": "Uploaded Natural Earth 10m Ports",
+                    "description": "Ports from Natural Earth",
+                    "source_operator": "OgrSource"
+                },
+                "meta_data": {
+                    "OgrMetaData": {
+                        "loading_info": {
+                            "file_name": "operators/test-data/vector/data/ne_10m_ports/ne_10m_ports.shp",
+                            "layer_name": "ne_10m_ports",
+                            "data_type": "MultiPoint",
+                            "time": "none",
+                            "columns": {
+                                "x": "",
+                                "y": null,
+                                "numeric": [
+                                    "natlscale"
+                                ],
+                                "decimal": [
+                                    "scalerank"
+                                ],
+                                "textual": [
+                                    "featurecla",
+                                    "name",
+                                    "website"
+                                ]
+                            },
+                            "default_geometry": null,
+                            "force_ogr_time_filter": false,
+                            "on_error": "skip",
+                            "provenance": null
+                        },
+                        "result_descriptor": {
+                            "data_type": "MultiPoint",
+                            "spatial_reference": "EPSG:4326",
+                            "columns": {
+                                "website": "Text",
+                                "name": "Text",
+                                "natlscale": "Number",
+                                "scalerank": "Decimal",
+                                "featurecla": "Text"
+                            }
+                        }
+                    }
+                }
+            }`,
+        },
+    ];
+
+    loadingInfo = '';
+
+    constructor(private dataSetService: DataSetService, private notificationService: NotificationService) {}
+
+    ngOnInit(): void {}
+
+    selectFiles(event): void {
+        this.selectedFiles = event.target.files;
+        console.log(this.selectedFiles);
+    }
+
+    upload(): void {
+        const form = new FormData();
+
+        for (const file of Array.from(this.selectedFiles)) {
+            form.append('files[]', file, file.name);
+        }
+
+        this.submittedUpload = true;
+
+        this.dataSetService.upload(form).subscribe(
+            (event) => {
+                if (event.type === HttpEventType.UploadProgress) {
+                    this.progress$.next(Math.round((100 * event.loaded) / event.total));
+                } else if (event.type === HttpEventType.Response) {
+                    this.uploadId$.next(event.body.id);
```

Comment:
don't you need to reset progress?

Reply:
I leave it at 100% to indicate that the upload is done. You can only upload once because it requires multiple steps (upload, loading info). You have to open the dialog again and then it's reseted